### PR TITLE
feat: Support for setting aggregation for (Hybrid)RouteLayer

### PR DESCRIPTION
**Pull request by andreped**
fixes https://github.com/aurelio-labs/semantic-router/issues/201.

Also fixed:
* Added unit tests to verify that aggregation selection updates the aggregation
* Added unit tests to verify that aggregation works as intended for both `RL` and `HybridRL`
* Added support to set `top_k` for `RouteLayer` similar to `HybridRouteLayer`
* Added unit tests for `top_k` for `RouteLayer`
* Added some error handling when setting both `top_k` and `aggregation`
* Linted code

### CODE REVIEW

**Comment by andreped** [comment]
Just realized that the `codecov.xml` file is inside the repo. IMO, it does not make any sense to have there. Code coverage is viewed in an external website, which then is linked to [this badge in the README](https://[camo.githubusercontent.com/e81acc0661212aa58ed65ed435acdff723e8bbba063375ebd77bd5790718a0b5/68747470733a2f2f636f6465636f762e696f2f67682f617572656c696f2d6c6162732f73656d616e7469632d726f757465722f67726170682f62616467652e7376673f746f6b656e3d48384f4f4d5632545546](https://camo.githubusercontent.com/e81acc0661212aa58ed65ed435acdff723e8bbba063375ebd77bd5790718a0b5/68747470733a2f2f636f6465636f762e696f2f67682f617572656c696f2d6c6162732f73656d616e7469632d726f757465722f67726170682f62616467652e7376673f746f6b656e3d48384f4f4d5632545546)), right?

So can it be deleted? I can do this now in this PR, if of interest, as well as add `codecov.xml` to `.gitignore`.

**Comment by andreped** [comment]
I also see that `codecov` complains about 5 lines above, but they are all related to error handling. Do I need to add these in tests or should I add `#nopragma` or similar?

File: semantic_router/layer.py [hunk 1]
```diff
@@ -391,6 +403,18 @@ def _retrieve(self, xq: Any, top_k: int = 5) -> List[dict]:
         scores, routes = self.index.query(vector=xq, top_k=top_k)
         return [{"route": d, "score": s.item()} for d, s in zip(routes, scores)]
 
+    def _set_aggregation_method(self, aggregation: str = "SUM"):
+        if aggregation == "SUM":
+            return lambda x: sum(x)
+        elif aggregation == "MEAN":
+            return lambda x: np.mean(x)
+        elif aggregation == "MAX":
+            return lambda x: max(x)
```
Comment (bruvduroiu) [inline semantic_router/layer.py:412]: nit: let's please use lowercase for the `aggregation` here to keep in line with [Pandas .agg function names](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.agg.html)

**Review by bruvduroiu — CHANGES_REQUESTED** [review]
Thank you for the contribution! I agree that only using `sum` tends to favor Routes with more utterances. 

Very small nitpick on the choice of naming the aggregation strings. We want to keep in line with the Pandas lowercase aggregation function names.

File: semantic_router/layer.py (hunk 1 above)
Comment (andreped) [reply semantic_router/layer.py:412]: I can change that now. I just followed TensorFlow's convention instead.

Comment (andreped) [reply semantic_router/layer.py]: I have changed it now.

**Comment by andreped** [comment]
Just ran tests with this in one of my projects. Works fine! :] But of course, as a `top_k` og 3 worked best for my project, the aggregation method was less important. Then again for higher values of `top_k` I got a change in performance.

**Review by jamescalam — APPROVED** [review]

**Review by bruvduroiu — APPROVED** [review]

**Pull request merged**